processor/src/main/java/org/mapstruct/ap/processor/MethodRetrievalProcessor.java

```diff
@@ -80,19 +93,60 @@ public int getPriority() {
         return 1;
     }
 
+    private List<SourceMethod> retrievePrototypeMethods(TypeMirror typeMirror, MapperConfig mapperConfig) {
+        if ( typeMirror == null || typeMirror.getKind() == TypeKind.VOID ) {
+            return Collections.emptyList();
+        }
+
+        TypeElement typeElement = asTypeElement( typeMirror );
+        List<SourceMethod> methods = new ArrayList<SourceMethod>();
+        for ( ExecutableElement executable : getAllEnclosedExecutableElements( elementUtils, typeElement ) ) {
+
+            ExecutableType methodType = typeFactory.getMethodType( typeElement, executable );
+            List<Parameter> parameters = typeFactory.getParameters( methodType, executable );
+            boolean containsTargetTypeParameter = SourceMethod.containsTargetTypeParameter( parameters );
+
+            // prototype methods don't have prototypes themselves
```

Comment:
Could you give an example for such an ambiguity?

IIUC, you allow multi-level config inheritance for methods declared on actual mapper interfaces (no matter whether those methods are defined in one interface or an interface and a super-interface). If there is an ambiguity, it needs to be explicitly resolved by specifying `@InheritConfiguration`. Is there any real difference to the case where the method to inherit the config from is not defined on the mapper but on a config class?

I think both cases should be handled equally. Defining methods on config classes should be a rare cases. It's only needed if a) the return type is an abstract class or b) you don't want that method be usable as mapping method. But in all other cases I'd expect methods to inherit config from on actual mappers.